Avoid RTPBuffer allocation

arjunshajitech opened the pull request:
#### Description

Previously, NewInterceptor validated the RTP buffer size by calling rtpbuffer.NewRTPBuffer
this resulted in unnecessary memory allocations during interceptor construction, even though
the buffer itself was not used at that stage.


#### Reference issue
Fixes #...

JoTurk reviewed (commented):
cool change, thank you.

arjunshajitech committed:
Avoid RTPBuffer allocation

Previously, NewInterceptor validated the RTP
buffer size by calling rtpbuffer.NewRTPBuffer
this resulted in unnecessary memory allocations
during interceptor construction, even though
the buffer itself was not used at that stage.

mengelbart commented on pkg/nack/receive_log.go:
```diff
@@ -20,19 +19,8 @@ type receiveLog struct {
-
-	if !correctSize {
-		return nil, fmt.Errorf("%w: %d is not a valid size, allowed sizes: %v", ErrInvalidSize, size, allowedSizes)
+	if err := rtpbuffer.IsBufferSizeValid(size); err != nil {
```
This now uses `rtpbuffer.IsBufferSizeValid` to check if we can create a `receiveLog`. What if we ever change the conditions for only one of them? Could we rename the `IsBufferSizeValid` function to `IsPowerOfTwo` and move it out of the `rtpbuffer` package? That way, it would be clear what it does, and if we change the conditions of either `receiveLog` or `RTPBuffer`, it would not affect the other.
